TPCQCVis/reports/TPC_AQC_Template_Period.ipynb

```diff
@@ -207,7 +207,110 @@
+    "    histograms (list): A list of distributions to compare against the mean.\n",
+    "    runList (list): A list of run numbers corresponding to the distributions in 'histograms'.\n",
+    "    draw_plot (bool): If True, draws the KL divergence plot. If False, only prints the outlier information.\n",
+    "\n",
+    "    Returns:\n",
+    "    hOutlier (ROOT.TH1F): The histogram of KL divergence values for each run (if draw_plot=True).\n",
+    "    cOutlier (ROOT.TCanvas): The canvas object containing the KL divergence plot (if draw_plot=True).\n",
+    "    \n",
+    "    The function will print the run number that has the highest KL divergence, indicating \n",
+    "    the most divergent (outlier) run.\n",
+    "    \"\"\"\n",
+    "    files = len(runList)\n",
+    "\n",
+    "    # Creating the canvas and histogram objects only if draw_plot is True\n",
+    "    if draw_plot:\n",
+    "        # Outlier canvas\n",
+    "        cOutlier = ROOT.TCanvas(\"cOutlier\", \"KL divergence\", 1100, 400)\n",
+    "        cOutlier.SetBottomMargin(0.15)\n",
+    "        cOutlier.SetGridx()\n",
+    "\n",
+    "        # Outlier distribution\n",
+    "        hOutlier = ROOT.TH1F(\"h_kl_divergence\", histogram_name + \";Run;KL divergence\", \n",
+    "                             files, 0, files)\n",
+    "        hOutlier.SetLineWidth(2)\n",
+    "        hOutlier.SetMarkerSize(1.5)\n",
+    "        hOutlier.SetMarkerStyle(ROOT.kFullCircle)\n",
+    "        hOutlier.SetStats(0)\n",
+    "\n",
+    "    # Building mean histogram\n",
+    "    meanHistogram = None\n",
+    "    for i,histogram in enumerate(histograms):\n",
+    "        if i == 0:\n",
+    "            meanHistogram = histogram.Clone(\"meanHistogram\")\n",
+    "        else:\n",
+    "            meanHistogram.Add(histogram)\n",
+    "        \n",
+    "        if i == files - 1:\n",
+    "            meanHistogram.Scale(1 / len(histograms))\n",
+    "\n",
+    "    kl_divergence_values = []\n",
+    "    max_kl_run_number = None\n",
+    "\n",
+    "    for i in range(len(histograms)):\n",
+    "        kl_divergence_value = calculate_kl_divergence(normalize_distribution(histograms[i]), normalize_distribution(meanHistogram))\n",
+    "        kl_divergence_values.append(calculate_kl_divergence(normalize_distribution(histograms[i]), normalize_distribution(meanHistogram)))\n",
+    "        if draw_plot:\n",
+    "            hOutlier.Fill(i, kl_divergence_value)\n",
+    "            hOutlier.GetXaxis().SetBinLabel(i + 1, runList[i])\n",
+    "        if i == 0:\n",
+    "            max_kl_divergence = kl_divergence_value\n",
+    "            max_kl_run_number = runList[i]\n",
+    "        elif kl_divergence_value > max_kl_divergence:  # Change this condition\n",
+    "            max_kl_divergence = kl_divergence_value\n",
+    "            max_kl_run_number = runList[i]  # Store the corresponding run number\n",
```

Comment:
```suggestion
    "        if kl_divergence_value > max_kl_divergence:  # Change this condition\n",
    "            max_kl_divergence = kl_divergence_value\n",
    "            max_kl_run_number = runList[i]  # Store the corresponding run number\n",
```
if you initiallize max_kl_divergence to e.g. -1 then you can save one if check

Reply:
I see!
Okay, I removed the first if conditional clause, and added a `max_kl_divergence` initilization to -inifitiny for safety (in case  a `kl_divergence_value` is mistakenly calculated with negative value, although it should not)